modified:   model/ftn/w3iorsmd.ftn  to write all restart files in a s…

yangfanglin opened the pull request:
…ub-directory restart_wave

            to enable wave breakpoint restart capability in GFS global workflow

yangfanglin committed:
modified:   model/ftn/w3iorsmd.ftn  to write all restart files in a sub-directory restart_wave
            to enable wave breakpoint restart capability in GFS global workflow

yangfanglin commented:
This update will force the wave model to write all restart files in a sub-directory restart_wave.  This update greatly simplifies the forecast script to avoid unnecessary I/O traffic and makes the script much cleaner.  I have tested this change in GFS.v16 parallels.  It worked perfectly fine.

aliabdolali commented:
@yangfanglin 
Thanks for the pull request.
There is an issue (https://github.com/NOAA-EMC/WW3/issues/219) for this development.
I assigned two reviewers for your pull request. 
Thanks,
AA

yangfanglin commented:
In https://github.com/NOAA-EMC/WW3/pull/223 I provided a temporary solution with hard-wired restart_wave subdirectory.  It is perfectly fine for me to use a more general approach Jessica proposed in https://github.com/NOAA-EMC/WW3/issues/219.  To move the GFS.v16 project forward, can we first commit https://github.com/NOAA-EMC/WW3/pull/223 to  branch production/GFS.v16, and update it to use the general solution at a later time ?   I believe production/GFS.v16 is only used by GFS.v16 project.   We need to test and prove the WAVE break-point restart capability before we update the forecast script to use the proposed new directory structure.

RobertoPadilla-NOAA commented:
Its good to go.

JessicaMeixner-NOAA commented:
So do we want to move forward with hard-coding this in the production/gfsv16 branch, knowing that this will have to go in as an option in the develop branch?  If so, I think we should remove the print* debug statements before merging this. 

If we do not hard-code the restart_wave folder name, this will require updates to https://github.com/NOAA-EMC/global-workflow/blob/feature/gfsv16b/scripts/exwave_prep.sh and/or the fix file fix_wave_gfs/ww3_multi.gfs.inp.tmpl

yangfanglin committed:
remove print statement

yangfanglin commented:
Jessica,

I have removed the print statement and updated
branch production/GFS.v16_restart.

Fanglin

On Wed, Jul 8, 2020 at 10:34 AM Jessica Meixner <notifications@github.com>
wrote:

> So do we want to move forward with hard-coding this in the
> production/gfsv16 branch, knowing that this will have to go in as an option
> in the develop branch? If so, I think we should remove the print* debug
> statements before merging this.
>
> If we do not hard-code the restart_wave folder name, this will require
> updates to
> https://github.com/NOAA-EMC/global-workflow/blob/feature/gfsv16b/scripts/exwave_prep.sh
> and/or the fix file fix_wave_gfs/ww3_multi.gfs.inp.tmpl
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/WW3/pull/223#issuecomment-655558659>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2NH3S3ZUD4B7DXO6KDR2R7XFANCNFSM4ORY3DRQ>
> .
>


-- 
Fanglin Yang, Ph.D.
Physical Scientist
Environmental Modeling Center
National Centers for Environmental Prediction
301-6833722; fanglin.yang@noaa.gov
http://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/
http://www.emc.ncep.noaa.gov/gmb/STATS_vsdb/

aerorahul commented:
This issue has re-emerged for GFSv17, GEFSv13 and SFSv1
https://github.com/NOAA-EMC/WW3/issues/1221 is probably a duplicate, but adds output of WW3 to be organized in a similar manner.
